cms/static/js/views/course_outline.js

```diff
@@ -209,6 +210,30 @@ define(["jquery", "underscore", "js/views/xblock_outline", "js/views/utils/view_
                         ensureChildrenRendered: this.ensureChildrenRendered.bind(this)
                     });
                 }
+            },
+
+            handleReIndexEvent: function(event) {
+                var self = this;
+                event.preventDefault();
+                var target = $(event.currentTarget);
+                target.css('cursor', 'wait');
+                this.startReIndex().done(function(data) {
+                    if(data.status !== 'success') {
+                        self.onIndexError(data.status);
+                    }
+                }).always(function() {target.css('cursor', 'pointer');});
+            },
+
+            startReIndex: function() {
+                var locator =  window.course.id;
+                return $.ajax({
+                    url: '/course_index/' + locator,
+                    method: 'GET'
+                    });
+            },
+
+            onIndexError: function(errors) {
+                alert(errors.join('\n'));
```

Comment:
This seems like an awkward way of notifying errors. Why not use the system feedback templates? Here's an example of showing an error prompt: https://github.com/edx/edx-platform/blob/master/cms/static/js/factories/manage_users.js#L15